execution/function/functions.go

```diff
@@ -4,64 +4,29 @@
 package function
 
 import (
-	"fmt"
 	"math"
 	"time"
 
-	"github.com/efficientgo/core/errors"
 	"github.com/prometheus/prometheus/model/histogram"
-
-	"github.com/thanos-io/promql-engine/execution/parse"
-	"github.com/thanos-io/promql-engine/parser"
 )
 
-var InvalidSample = Sample{T: -1, F: 0}
+var invalidSample = sample{T: -1, F: 0}
 
-type Sample struct {
+type sample struct {
 	T int64
 	F float64
 	H *histogram.FloatHistogram
 }
 
-type FunctionArgs struct {
-	Samples          []Sample
-	StepTime         int64
-	SelectRange      int64
-	ScalarPoints     []float64
-	Offset           int64
-	MetricAppearedTs *int64
+type functionArgs struct {
+	Samples      []sample
+	StepTime     int64
```

Comment:
Do we need to use this parameter in functions?

Reply:
ah no, we only need it in range functions i think, not removing it was an oversight

Reply:
ah its somewhat used in the time related functions; most of the time not though

Reply:
i propose that we keep it for now and maybe handle the time based functions with a special sub operator here, wdyt?

Reply:
Sounds good to me 👍

Reply:
i went ahead and removed it too because it would complicate the date functions. i now introduced noArgs and withArgs variants of them and got rid of step time most of the functions that way.